headers-ext/src/common/content_type.rs

```diff
@@ -0,0 +1,137 @@
+use mime::{self, Mime};
+
+/// `Content-Type` header, defined in
+/// [RFC7231](http://tools.ietf.org/html/rfc7231#section-3.1.1.5)
+///
+/// The `Content-Type` header field indicates the media type of the
+/// associated representation: either the representation enclosed in the
+/// message payload or the selected representation, as determined by the
+/// message semantics.  The indicated media type defines both the data
+/// format and how that data is intended to be processed by a recipient,
+/// within the scope of the received message semantics, after any content
+/// codings indicated by Content-Encoding are decoded.
+///
+/// Although the `mime` crate allows the mime options to be any slice, this crate
+/// forces the use of Vec. This is to make sure the same header can't have more than 1 type. If
+/// this is an issue, it's possible to implement `Header` on a custom struct.
+///
+/// # ABNF
+///
+/// ```text
+/// Content-Type = media-type
+/// ```
+///
+/// # Example values
+///
+/// * `text/html; charset=utf-8`
+/// * `application/json`
+///
+/// # Examples
+///
+/// ```
+/// # extern crate headers_ext as headers;
+/// use headers::ContentType;
+///
+/// let ct = ContentType::json();
+/// ```
+#[derive(Clone, Debug, PartialEq)]
+pub struct ContentType(Mime);
```

Comment:
Is there no way to get the actual value in there or to build a content type that isn't in the like 5 below :/ ?

Reply:
Oh nvm, I can use the `From` impl. Would still be nice to get the opposite `Into<Mime>`

Reply:
Definitely need it

```rust
mString = match headers.typed_get::<ContentType>() {
    Some(ContentType(ref mime)) => mime.to_string(),
    _ => "".to_owned()                              
};
```

Reply:
That one at least looks like `ContentType` should implement `Display`. But there should also be a away to get the Mime... I believe the Mime crate should split Mime into MediaType and MediaRange...